File: python-pr/52/python/code-quality-readability-maintainability/url_builder_21.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def build_url(base_url, path=None, params=None):
    url = base_url
    if path:
        if not url.endswith('/') and not path.startswith('/'):
            url += '/'
        url += path
    if params:
        url += '?'
        param_strings = []
        for key, value in params.items():
            param_strings.append(f"{key}={value}")
        url += '&'.join(param_strings)
    return url
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/url_builder_21.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function buildUrl(baseUrl, path = null, params = null) {
    let url = baseUrl;
    if (path) {
        if (!url.endsWith('/') && !path.startsWith('/')) {
            url += '/';
        }
        url += path;
    }
    if (params) {
        url += '?';
        let paramStrings = [];
        for (let [key, value] of Object.entries(params)) {
            paramStrings.push(`${key}=${value}`);
        }
        url += paramStrings.join('&');
    }
    return url;
}
// {/fact}